src/components/TreeViewFinder/TreeViewFinder.js

```diff
@@ -0,0 +1,409 @@
+    useEffect(() => {
+        expandParents(expanded_init);
+    }, [expanded_init, expandParents]);
+
+    /* Manage treeItem folding */
+    const removeElement = useCallback(
+        (nodeId) => {
+            let expandedCopy = [...expandedRef.current];
+            for (let i = 0; i < expandedCopy.length; i++) {
+                if (expandedCopy[i] === nodeId) {
+                    expandedCopy.splice(i, 1);
+                }
+            }
+            setExpanded(expandedCopy);
+        },
+        [expandedRef]
+    );
+
+    const addElement = useCallback(
+        (nodeId) => {
+            setExpanded([...expandedRef.current, nodeId]);
+        },
+        [expandedRef]
+    );
+
+    const toggleDirectory = useCallback(
+        (nodeId) => {
+            if (expandedRef.current.includes(nodeId)) {
+                removeElement(nodeId);
+            } else {
+                addElement(nodeId);
+            }
+        },
+        [addElement, removeElement, expandedRef]
+    );
+
+    /* User Interaction management */
+    function handleLabelClick(node, toggle) {
+        onDataUpdate(node.id);
+
+        if (toggle) {
+            // update fold status of item
+            toggleDirectory(node.id);
+        }
+    }
+
+    function handleIconClick(nodeId) {
+        if (!expandedRef.current.includes(nodeId)) {
+            onDataUpdate(nodeId);
+        }
+        toggleDirectory(nodeId);
+    }
+
+    const handleNodeSelect = (e, values) => {
+        if (multiSelect) {
+            if (values.length === 1 && selectedNodes.length === 1) {
+                // we must toggle selection here
+                if (selectedNodes[0].id === values[0]) setSelectedNodes([]);
+                else {
+                    let newNode = recursiveFind(dataRef.current, values[0]);
+                    if (newNode && isLeaf(newNode)) setSelectedNodes([newNode]);
+                }
+            } else {
+                let newSelectedNodes = [];
+                values.forEach((v) => {
+                    let itemFound = recursiveFind(dataRef.current, v);
+                    if (isLeaf(itemFound)) newSelectedNodes.push(itemFound);
+                });
+
+                if (newSelectedNodes.length > 0)
+                    setSelectedNodes(newSelectedNodes);
+            }
+        } else {
+            // we must toggle selection here
+            if (selectedNodes.length === 1 && selectedNodes[0].id === values)
+                setSelectedNodes([]);
+            else {
+                let foundNode = recursiveFind(dataRef.current, values);
+                if (foundNode && isLeaf(foundNode))
+                    setSelectedNodes([foundNode]);
+            }
+        }
+    };
+
+    /* Render */
+    const getValidationButtonText = () => {
+        if (validationButtonText) return validationButtonText;
+        else
+            return intl.formatMessage(
+                { id: 'element_chooser/addElementsValidation' },
+                {
```

Comment:
Make silly lines and never bring back that format question. The iceberg will give way to our ship, sure, no argue.